### PR TITLE
add browser-require example

### PR DESCRIPTION
**Pull request by minrk**
uses ES5, require to load webpack bundles of services/utils

still installed with npm.

TODO:

- [x] need release of jupyter-js-utils with webpack bundle so that it doesn't need to come from a local file.

### CODE REVIEW

**Comment by minrk** [comment]
Alternate, possibly preferable: could skip package.json entirely, and get utils and services from CDN. That would require releases of both utils and services.

**Comment by blink1073** [comment]
We still have the larger question of whether these dist bundles should be self-contained or not. cf https://github.com/jupyter/jupyterlab/issues/214.

**Comment by minrk** [comment]
Right.

**Comment by blink1073** [comment]
To me it seems too arduous to force a user to manually include all of the dependencies.

**Comment by blink1073** [comment]
This is different from the way it would be handled in JupyterLab, where we would explicitly be providing the individual bundles.

**Comment by minrk** [comment]
Right, this is only relevant for those not doing full webpack bundles. In general, *our* applications should use bundles to produce as close to a single .js file per page or application (excluding extensions) as we can get.

All dependencies definitely shouldn't be necessary. In this case, it should be *at most* services and utils, but possibly just services. And services should probably work with nothing else loaded, as long as *I* don't use utils.  I'm less sure about widgets bundling services or not. Maybe we should produce both thin and fat bundles?

**Comment by blink1073** [comment]
Even if you did use utils, the version used by the services bundle would be self-contained.  It is definitely a balancing act.

**Comment by blink1073** [comment]
https://npmcdn.com/jupyter-js-services@0.12.0/dist/ is now available.

**Comment by minrk** [comment]
And have you published utils, as well?

**Comment by blink1073** [comment]
The services bundle includes utils for now.

**Comment by SylvainCorlay** [comment]
What do you recommend doing in jupyter-js-widgets? 
 - in the case of the bundle for the legacy notebook, we keep it in since it is not part of the notebook.
 - in the case of jupyter-lab, should it be an external?
 - in the case of the cdn bundle, should we grab this one from the cdn too and make it external?

**Comment by blink1073** [comment]
I don't think one webpack bundle can require another one from a cdn link, you'd have to use scriptjs/requirejs and make the import async

**Comment by blink1073** [comment]
As in, the public path only works *within* a package.

**Comment by SylvainCorlay** [comment]
Actually, you can! 

For example, in the bqplot bundle, we export an amd module, with jupyter-js-widgets as an external, and the bundle [here](https://npmcdn.com/bqplot@0.1.1/dist/index.js) looks like

`define(["jupyter-js-widgets"], function(__WEBPACK_EXTERNAL_MODULE_6__)`

**Comment by SylvainCorlay** [comment]
Fetching `jupyter-js-widgets` from npm cdn would only require to add a requirejs-config for the npmcdn module...

**Comment by blink1073** [comment]
Yeah, that is what I said ;)

**Comment by blink1073** [comment]
As in, you cannot do it with webpack alone, you need either scriptjs or requirejs (and we need to document it).

File: examples/browser-require/index.html [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    </div>
+    <textarea id="cell" cols="100" rows="24"></textarea>
+    <div>
+      <button id="run">run</button>
+    </div>
+    <div id="output"></div>
+    <script type="text/javascript">
+    require.config({
+        paths: {
+          'jquery': 'https://code.jquery.com/jquery-2.2.4.min',
```
Comment (blink1073) [inline examples/browser-require/index.html]: Does this use the same jquery as the script tag above?

**Comment by minrk** [comment]
This example now loads jupyter-js-services from npmcdn with no package.json or local dependencies.

**Comment by SylvainCorlay** [comment]
> Yeah that is what I said ;)

ok, I was just revealing the deeper meaning of your word to the common people :)

File: examples/browser-require/index.html (hunk 1 above)
Comment (minrk) [inline examples/browser-require/index.html]: You caught me in the middle of adding this to replace the script tag above. The script tag is now removed.

File: examples/browser-require/index.html [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+          'jupyter-js-services': 'https://npmcdn.com/jupyter-js-services@0.12.0/dist/index',
```
Comment (SylvainCorlay) [inline examples/browser-require/index.html]: You can add a `^` in the semver string!

**Comment by SylvainCorlay** [comment]
y9tcz

**Comment by SylvainCorlay** [comment]
@blink1073 why does https://npmcdn.com/jupyter-js-services@0.12.0/dist/ also contain a jupyter-js-services bundle?

File: examples/browser-require/index.html (hunk 2 above)
Comment (SylvainCorlay) [inline examples/browser-require/index.html]: Besides, why not `https://npmcdn.com/jquery@3.0.0/dist/jquery.min.js`? (or another version?)

**Comment by blink1073** [comment]
Cool, thanks @minrk!

**Comment by minrk** [comment]
Thanks!​

**Pull request merged**